src/rng.rs

```diff
@@ -393,14 +393,20 @@ impl Fill for [u8] {
     }
 }
 
-macro_rules! impl_fill {
+// This macro is unsafe to call: target types must support transmute from
+// random bits (i.e. all bit representations are valid).
+macro_rules! unsafe_impl_fill {
```

Comment:
Would it be possible for the various integer types to implement a "Safe to mem::transmute from raw bytes" sealed marker trait, and then implement this via a generic impl instead of a macro?